CMakeLists.txt

```diff
@@ -13,26 +13,17 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # Compile comamnds for clang tools
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
-# Third party deps
-set(THIRD_PARTY_INSTALL_DIR ${CMAKE_BINARY_DIR}/third-party)
-
 # Library funcs
-if (BUILD_SHARED_LIBS)
-    function(faabric_public_lib lib_name lib_deps)
-        add_library(${lib_name} SHARED ${lib_deps})
-    endfunction()
-    function(faabric_private_lib lib_name lib_deps)
-        add_library(${lib_name} STATIC ${lib_deps})
+function(faabric_lib lib_name lib_deps)
+    add_library(${lib_name} ${lib_deps})
+
+    add_library(${lib_name}_obj OBJECT ${lib_deps})
+
+    if(BUILD_SHARED_LIBS)
```

Comment:
This function is just for building libraries internally, so we want to add the normal static library for linking between internal libraries as normal, then the `OBJECT` library to support bundling everything together at the end.